File: core/src/commonMain/kotlin/com/niceoneconsent/sdk/models/Purpose.kt
package com.niceoneconsent.sdk.models

import kotlinx.serialization.Serializable

/**
 * Represents a single consent purpose fetched from the backend.
 *
 * @property id Unique identifier for the purpose (e.g., "sales", "marketing")
 * @property name Display name of the purpose
 * @property description Human-readable description of what this purpose entails
 * @property mandatory Whether this purpose is required and cannot be deselected
 */
@Serializable
data class Purpose(
    val id: String,
    val name: String,
    val description: String,
    val mandatory: Boolean = false
)

/**
 * Wrapper for the purposes API response.
 */
@Serializable
data class PurposesResponse(
    val purposes: List<Purpose>
)

File: ios-sdk/Sources/ConsentSDK/Models/Purpose.swift
import Foundation

/// Represents a single consent purpose fetched from the backend.
public struct Purpose: Codable, Identifiable, Equatable {
    /// Unique identifier for the purpose (e.g., "sales", "marketing")
    public let id: String
    /// Display name of the purpose
    public let name: String
    /// Human-readable description of what this purpose entails
    public let description: String
    /// Whether this purpose is required and cannot be deselected
    public let mandatory: Bool

    public init(id: String, name: String, description: String, mandatory: Bool = false) {
        self.id = id
        self.name = name
        self.description = description
        self.mandatory = mandatory
    }
}

/// Wrapper for the purposes API response.
struct PurposesResponse: Codable {
    let purposes: [Purpose]
}
